### ISSUE TITLE
Error on Ubuntu when unpacking data file

### ISSUE THREAD

**Issue by rohitchormale** [mrqc/bitcoin-blk-file-reader#3]
Thanks for sharing program. When I was trying to run program on Ubuntu, it is giving below error

`unpack requires a string argument of length 4. `

at this func 

```
def readIntLittleEndian(blockFile):
  return struct.pack(">I", struct.unpack("<I", blockFile.read(4))[0])


```

Can you please let me know which os, is it tested ?

**Comment by rohitchormale** [comment]
I think i figured it out. will do more testing and will send patch soon.

**Comment by mrqc** [comment]
Hi! sorry did not get any notification for your issue. I was testing it on ubuntu with python2.7

**Comment by mrqc** [comment]
Maybe your bug is fixed with the latest version?

**Comment by rohitchormale** [comment]
Yes I tested it on Mint 19 which is based on Ubuntu 18.04. If you want to test any other os, before merging, let me know. happy to assist.

**Comment by mrqc** [comment]
Your version or my latest version. Because I think your implementation does not consider extended transaction format. This was the bug.

**Comment by rohitchormale** [comment]
i tested the latest code. infinite loop issue is fixed. so you can close issue and pull request.

**Comment by rohitchormale** [comment]
i tested the latest code. infinite loop issue is fixed. so you can close issue and pull request. thnx again man for this extractor. it really helped one of my friend from uni.
